src/UI/class-settings-page.php

```diff
@@ -1141,15 +1143,15 @@ private function validate_advanced_section( $input ) {
 			$input['metadata_secret'] = '';
 		} else {
 			$input['metadata_secret'] = $this->get_unobfuscated_value( $input['metadata_secret'], $this->parsely->get_options()['metadata_secret'] );
-			if ( '' !== $input['metadata_secret'] ) {
-				if ( strlen( $input['metadata_secret'] ) !== 10 ) {
-					add_settings_error(
-						Parsely::OPTIONS_KEY,
-						'metadata_secret',
-						__( 'The Metadata Secret was not saved because it is incorrect. Please contact Parse.ly support!', 'wp-parsely' )
-					);
-					$input['metadata_secret'] = $options['metadata_secret'];
-				}
+			$metadata_secret_length   = strlen( $input['metadata_secret'] );
+			if ( $metadata_secret_length > 0 &&
+					false === Validator::validate_metadata_secret( $input['metadata_secret'] ) ) {
```

Comment:
`$metadata_secret_length > 0` is unnecessary here. That's the job of the validator now.

```suggestion
			if ( false === Validator::validate_metadata_secret( $input['metadata_secret'] ) ) {
```

Reply:
We could make the validation function accept an empty value as valid through an `$allow_empty` argument, though I would not do that as a default behavior (since an empty value cannot really be a valid secret value). This is why this piece was left there, but I understand the reasoning.